workflow/controller/operator_concurrency_test.go

```diff
@@ -486,3 +487,331 @@ func TestSynchronizationWithRetry(t *testing.T) {
 
 	})
 }
+
+const StepWithSync = `
```

Comment:
these examples are a bit long for me to understand - could you please simplify them are some how make them easier to understand? maybe add comments?